### ISSUE TITLE
search result aria label should prioritze content over location

### ISSUE THREAD

**Issue by meganrogge** [microsoft/vscode#179718]
reported by @jooyoungseo, the aria label for search results should read content first then location IE line/column info

![A screen shot of the search results](https://user-images.githubusercontent.com/29464607/231282680-460d901e-c5e6-495c-ae55-774ce15da4bd.png)

cc @isidorn for historical context

**Comment by isidorn** [comment]
Makes good sense, and should be an easy fix.

**Comment by andreamah** [comment]
When replace is disabled, we have a label that looks like 
```
Found '{match}' at column {columnNum} in line '{wholeLineSurroundingMatch}'
```
Otherwise, if replace is enabled, we have:
```
Replace '{match}' with '{replaceText}' at column {columnNum} in line {wholeLineSurroundingMatch}
```

https://github.com/microsoft/vscode/blob/a47c78516c0b314d3766c2a2cc597dd582e080a2/src/vs/workbench/contrib/search/browser/searchResultsView.ts#L403-L407

Would you rather see something like 
```
Found '{match}' in line '{wholeLineSurroundingMatch}' at column {columnNum} 
```
for non-replace and 
```
Replace '{match}' with '{replaceText}' in line {wholeLineSurroundingMatch} at column {columnNum} 
```
for replace?


> Interesting - this works as expected on macOS

Not too sure why this would be different on mac. What does the aria say on your machine?

**Comment by meganrogge** [comment]
@andreamah I have not tried on Windows, but when @jooyoungseo shared his screen, the announcement read the line/column info before the actual content. Do you know where that would be coming from?

**Comment by andreamah** [comment]
should be fixed in #179795

**Comment by meganrogge** [comment]
@andreamah have you tested this on  windows? Your fix changes the position of line vs column, which is good, but what JooYoung reported was that those preceded the actual content on windows.

**Comment by meganrogge** [comment]
I just tried on Windows and cannot reproduce the behavior that @jooyoungseo demoed - in which the line/col info was presented before the actual content. We will need to hear from him if it still happens and if so, how to reproduce.

**Comment by andreamah** [comment]
> I just tried on Windows and cannot reproduce the behavior that @jooyoungseo demoed - in which the line/col info was presented before the actual content. We will need to hear from him if it still happens and if so, how to reproduce.

Did you just try on insiders? I just changed the functionality

**Comment by jooyoungseo** [comment]
@meganrogge and @andreamah 

## Current Behavior

> 3 matches in file bar.txt of folder foo

## Suggested Behavior

> foo/bar.txt 3 matches

## Current Behavior

> Found 'VSCode' in line 'VSCode is cool' at column 15 

## Suggested Behavior

> 'VSCode is cool' in line  at column 15 

* Note: `Found ’VSCode’ in` is redundant b/c that's the user input search term and users already know what they are looking for, and it takes longer time for screen reader users to hear the actual content.

**Comment by andreamah** [comment]
> Note: Found ’VSCode’ in is redundant b/c that's the user input search term and users already know what they are looking for, and it takes longer time for screen reader users to hear the actual content.

The reason that we state the match again is for regex users. If they have a fairly general regex, they might want to know what actually caused it to match. Although we could also argue that the line context is more commonly valued by search users, since many people don't use regex.

**Comment by meganrogge** [comment]
Thanks for that context @andreamah.  I agree that we should optimize for the most common case here.

**Comment by jooyoungseo** [comment]
@meganrogge -- I recommend taking out redundant prefix, including 'Replace'. This is similar to one of the recommended best practices creating alt text. We recommend avoiding any redundant words, such as "Image of " or "Graphics of  " b/c users know what they are interacting with.

**Comment by andreamah** [comment]
Using the example from before, we could probably do something like 

'VSCode is cool' found in match 'VSCode' in line at column 15

when replace is not active. What do you think the replace case should look like?

**Comment by meganrogge** [comment]
Although I'm thinking we might not even need `'VSCode' in ` as a prefix here since as @jooyoungseo pointed out, the user entered the search term and thus knows what it is already

**Comment by andreamah** [comment]
Since the replace hasn't occurred at this point yet, `replaced` wouldn't be past tense in the results. We can say, `will be replaced` instead, but is that too wordy?

Also, I feel that the match string `VSCode` is still useful here for those who search for regex, so perhaps that could go at the end? Or we could just read it if they have regex search enabled.

**Comment by andreamah** [comment]
fyi, 'm going to move this to that may iteration because I will be offline for the rest of the April iteration.

**Comment by meganrogge** [comment]
I might fix it for you 😄

**Comment by amunger** [comment]
It sounds like line context is still expected from the comments here, but only the column number is stated.

e.g.
'matched context' at column 34 found (1) 1 of 1

Also, finding a match in the first column will state that it was found at column 2

**Comment by meganrogge** [comment]
@amunger  I see why you were confused but `line` is text not a line number there. So this is working as designed.

**Comment by meganrogge** [comment]
not sure what you mean by this

> Also, finding a match in the first column will state that it was found at column 2

**Comment by amunger** [comment]
created https://github.com/microsoft/vscode/issues/180982 since it seems like a different issue
